src/main/java/br/unicamp/meca/models/ActionSequencePlan.java

```diff
@@ -44,22 +44,36 @@ public ActionSequencePlan(ActionStep[] actionIdSequence) {
 	}
 	
 	/**
-	 * Returns the id of the current ActionFromPlanningCodelet to be undertaken in the plan.
+	 * Returns the ActionStep of the current ActionFromPlanningCodelet to be undertaken in the plan.
 	 * 
 	 * @return currentActionId
 	 */
 	public ActionStep getCurrentActionStep() {
 		return actionIdSequence[currentActionIdIndex];
 	}
+        
+        /**
+	 * Returns the ActionStep of the last ActionFromPlanningCodelet to be undertaken in the plan, or null if it is the first step.
+	 * 
+	 * @return currentActionId
+	 */
+	public ActionStep getLastActionStep() {
+            if (currentActionIdIndex == 0) return(actionIdSequence[actionIdSequence.length-1]);
+            else return actionIdSequence[currentActionIdIndex-1];
```

Comment:
Isn't the last step to be undertaken always `actionIdSequence.length-1` ?

Reply:
The idea of LastActionStep is to return always currentActionIdIndex -1, i.e. this is last in terms of "last executed", not last as in "last of the plan". There is a special case, which is when actionIdIndex is equal to 0. It may be the first execution of the plan, when in this case there is no "last executed", as it is the first step of the plan. Or, it may be the case the plan has finished and came back to actionIdIndex=0. To differentiate among these two, there is the flag "executed" in the ActionStep. If 0 means the first time a plan is executed, then the flag "executed" is false. If 0 means the plan has already been executed and rotated back to 0, then the flat "executed" is true.

Reply:
Shouldn't this method be called `getLastExecutedActionStep`, then?